src/naemon/objects.c

```diff
@@ -109,7 +124,41 @@ int fcache_objects(char *cache_file)
 			fcache_hostescalation(fp, esclist->object_ptr);
 	}
 
-	fclose(fp);
+	/* reset file permissions */
+	fchmod(fd, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH);
+
+	/* flush the file to disk */
+	fflush(fp);
+
+	/* fsync the file so that it is completely written out before moving it */
+	fsync(fd);
+
+	/* close the temp file */
+	result = ferror(fp) | fclose(fp);
+
+	/* save/close was successful */
+	if (result == 0) {
+
+		result = OK;
+
+		/* move the temp file to the status log (overwrite the old status log) */
```

Comment:
Just a minor comment here. I guess status log should be cache file.

Reply:
indeed, missed that while copy/pasting from the status.dat code.

Reply:
should be fine now